akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToResponseMarshallers.scala

```diff
@@ -118,7 +126,14 @@ trait LowPriorityToResponseMarshallerImplicits {
                   case Marshalling.WithOpenCharset(`bestMT`, marshal)    ⇒ () ⇒ marshal(bestCS)
                 }
             }).toList
+
+            // TODO we could either special case for certrain known types,
+            // or extend the entity support to be more advanced such that it would negotiate the element content type it
```

Comment:
We recently created https://github.com/akka/akka-http/issues/535, which also discussed possibly making the algorithm to select a marshaller more content-type-aware. Although that ticket is about matching the incoming `Accept:` header, not just the scala implicits.